cpp/CMakeLists.txt

```diff
@@ -534,19 +536,73 @@ if(BUILD_CUML_CPP_LIBRARY)
       $<INSTALL_INTERFACE:include>
   )
 
+  set(_cuml_cpp_public_libs)
+  set(_cuml_cpp_private_libs)
+
+  if(CUML_USE_RAFT_STATIC AND (TARGET raft::raft))
+    get_target_property(_includes raft::raft INTERFACE_INCLUDE_DIRECTORIES)
+    target_include_directories(${CUML_CPP_TARGET} PUBLIC ${_includes})
```

Comment:
why are these changes to the includes lists required like this? Explanation comments would be good since this is more complex to understand than before. Some targets like treelite have explanations which are great, but things like RAFT here don't

Reply:
If we're statically linking RAFT, the RAFT headers need to be added to the PUBLIC interface of the cuML target, even if the static lib linkage is marked PRIVATE. This ensures the Cython objects can still `-llibcuml++`, without needing to `-llibraft_nn_static`.

Reply:
I've extracted the interface export into a function to better document what we're doing in all the places where we're extracting and copying interface includes.